### PR TITLE
feat: Generate client upload token

### LINKED ISSUE #2: Creation of client upload route
**User story**:
As a user I want to be able to upload files for safekeeping in the cloud, so that I don't risk loosing them.

**See:**
https://vercel.com/docs/storage/vercel-blob/quickstart#create-a-client-upload-route

### PR DESCRIPTION
**Pull request by frederikpyt**
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -28,4 +28,7 @@ REDIS_HOST="redis"
 REDIS_PORT="6379"
 REDIS_USER=""
 REDIS_PASSWORD=""
-REDIS_URL="rediss://${REDIS_USER}:${REDIS_PASSWORD}@${REDIS_HOST}:${REDIS_PORT}"
\ No newline at end of file
+REDIS_URL="rediss://${REDIS_USER}:${REDIS_PASSWORD}@${REDIS_HOST}:${REDIS_PORT}"
```
Comment (kristianbinau) [inline .env.example]: Can you please just change `REDIS_HOST` to `localhost` & this to `redis` instead of `rediss`

File: .env.example [hunk 2]
```diff
@@ -28,4 +28,7 @@ REDIS_HOST="redis"
...
+
+# Vercel Blob Storage
+BLOB_READ_WRITE_TOKEN=""
```
Comment (kristianbinau) [inline .env.example]: Write Docs how to setup shiii

File: prisma/schema.prisma [hunk 3]
```diff
@@ -12,25 +12,54 @@ datasource db {
...
+    updatedAt DateTime @updatedAt
 
     @@unique([tokenFamily])
     @@index([userId])
     @@index([userId, tokenFamily])
 }
+
+model Item {
+    id       Int    @id @default(autoincrement())
+    name     String @db.VarChar(50)
```
Comment (kristianbinau) [inline prisma/schema.prisma]: Longer

File: src/modules/item/blob.service.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+				return {
+					allowedContentTypes,
+					tokenPayload: formatTokenPayload
+						? await formatTokenPayload(clientPayload, accessTokenPayload)
+						: undefined,
+				};
+			},
+			onUploadCompleted,
+		});
```
Comment (kristianbinau) [inline src/modules/item/blob.service.ts]: Leave some good small comments explaining stuff

Comment (kristianbinau) [reply src/modules/item/blob.service.ts]: And maybe prettier Typescript

File: .env.example (hunk 2 above)
Comment (kristianbinau) [inline .env.example]:
```suggestion
BLOB_READ_WRITE_TOKEN="vercel_blob_rw_0511953119b0c1167283c7453a088727" # Fake vercel token
```

File: src/modules/item/item.schema.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		blob: z
+			.object({
+				url: z.string(),
+				pathname: z.string(),
+				contentType: z.string(),
+				contentDisposition: z.string(),
+			})
+			.optional(),
+	}),
+});
```
Comment (kristianbinau) [inline src/modules/item/item.schema.ts]: Implement the Anders way!

**Review by kristianbinau — CHANGES_REQUESTED** [review]

File: src/modules/item/item.schema.ts (hunk 5 above)
Comment (frederikpyt) [reply src/modules/item/item.schema.ts]: 😭

File: src/modules/item/blob.service.ts (hunk 4 above)
Comment (frederikpyt) [reply src/modules/item/blob.service.ts]: Better?
image

**Review by kristianbinau — APPROVED** [review]
🪨

**Review by kristianbinau — CHANGES_REQUESTED** [review]
Failing tests

**Review by kristianbinau — APPROVED** [review]
🚀

**Pull request merged**